balena/models/hup.py

```diff
@@ -15,39 +17,52 @@ def get_hup_action_type(self, device_type, current_version, target_version):
         """
 
         try:
-            parsed_current_ver = semver.parse(current_version)
+            parsed_current_ver = Version.parse(current_version)
         except Exception:
             raise exceptions.OsUpdateError("Invalid current balenaOS version")
 
         try:
-            parsed_target_ver = semver.parse(target_version)
+            parsed_target_ver = Version.parse(target_version)
         except Exception:
             raise exceptions.OsUpdateError("Invalid target balenaOS version")
 
-        if parsed_current_ver["prerelease"] or parsed_target_ver["prerelease"]:
-            raise exceptions.OsUpdateError("Updates cannot be performed on pre-release balenaOS versions")
+        if parsed_current_ver.prerelease or parsed_target_ver.prerelease:
+            raise exceptions.OsUpdateError(
+                "Updates cannot be performed on pre-release balenaOS versions"
+            )
 
-        releases = (
-            self.__xstr(parsed_current_ver["prerelease"])
-            + self.__xstr(parsed_target_ver["prerelease"])
-            + self.__xstr(parsed_target_ver["build"])
-            + self.__xstr(parsed_current_ver["build"])
-        )
+        cur_variant = self.__get_variant(parsed_current_ver)
+        target_variant = self.__get_variant(parsed_target_ver)
 
-        if "dev" in releases:
-            raise exceptions.OsUpdateError("Updates cannot be performed on development balenaOS variants")
+        if target_variant is not None and (
+            (cur_variant == "dev") != (target_variant == "dev")
```

Comment:
I would say let's split this tiny change out of this PR and maybe even ship it in the current major.